app/assets/javascripts/alert-display.coffee

```diff
@@ -155,16 +155,15 @@ class AlertDisplay
       false
     )
 
-    # coffeelint: disable=max_line_length
-    widgetController.ractive.on('*.nlw-notify',          (_, message)           => @reportNotification(message))
-    widgetController.ractive.on('*.nlw-runtime-error',   (_, source, exception) => @reportRuntimeError(source, exception, widgetController.ractive.get('code')))
-    widgetController.ractive.on('*.nlw-compiler-error',  (_, source, errors)    => @reportCompilerErrors(source, errors))
-    widgetController.ractive.on('*.nlw-extension-error', (_, messages)          => @reportError(messages.join('<br/>')))
-    # coffeelint: enable=max_line_length
     return
 
-  # (String, Exception) => Unit
-  reportRuntimeError: (source, exception, code) ->
+  # (CommonEventArgs, { messages: Array[String] }) => Unit
+  'extension-error': (_, {messages}) ->
+    @reportError(messages.join('<br/>'))
+    return
+
+  # (CommonEventArgs, { source: String, exception: Exception }) => Unit
+  'runtime-error': (_, {source, exception}) ->
```

Comment:
I assume that you have a very good reason for it, but why use dasherized names here, instead of camel-cased ones?

Reply:
It's because they're events, essentially, and our event name convention has been kebab case.